src/Handlers/JsonPostHandler.php

```diff
@@ -39,6 +39,11 @@ protected function callAction(HttpRequest $request, callable $method, ?string $r
         $input = $this->readRequest($request, $requestClass);
         $output = call_user_func($method, $input);
         unset($input);
+
+        // in case of a direct response we skip further request handling and writing output again
+        if (in_array(JsonServiceHandler::DIRECT_RESPONSE, $request->tags))
```

Comment:
It is not documented yet, but request tags are key-value hash. So we should use isset() or array_key_exists().

Better yet, add setTag(), getTag() and hasTag() methods to HttpRequest directly (leaving public access to $tags as we do for other request properties).